deployment/init_grid/cloudformation/Makefile

```diff
@@ -20,24 +23,34 @@ $(BUILD_DIR)/grid_init/tag.$(TAG):
 	@touch $(BUILD_DIR)/grid_init/tag.$(TAG)
 
 $(BUILD_DIR)/grid_init/init-grid-$(TAG): $(BUILD_DIR)/grid_init/tag.$(TAG) $(YAML_SRC)
-	aws cloudformation create-stack --stack-name $(TAG) --region $(REGION) --template-body file://$(YAML_SRC) --parameters ParameterKey=BucketTag,ParameterValue=$(TAG)
-	@echo "Waiting for cloud formation successful deployment"
+	@if [ "$(shell aws cloudformation list-stacks --query 'StackSummaries[?StackStatus==`CREATE_COMPLETE` && StackName==`$(TAG)`].StackName' --output text)" != "$(TAG)" ]; then \
+	echo "CloudFormation stack $(TAG) doesn't exist. Creating ..."; \
+	aws cloudformation create-stack --stack-name $(TAG) --region $(REGION) --template-body file://$(YAML_SRC) --parameters ParameterKey=BucketTag,ParameterValue=$(TAG); else \
+	echo "Failed creating the CloudFormation stack as it may already exist. Trying to update ..."; \
+	aws cloudformation update-stack --stack-name $(TAG) --region $(REGION) --template-body file://$(YAML_SRC) --parameters ParameterKey=BucketTag,ParameterValue=$(TAG); fi
+	@echo "Waiting for the successful deployment of CloudFormation stack $(TAG) ..."
 	@aws cloudformation wait stack-create-complete --stack-name $(TAG) --region $(REGION)
 	@touch $(BUILD_DIR)/grid_init/init-grid-$(TAG)
-	aws cloudformation describe-stacks --stack-name $(TAG) --region $(REGION) --query 'Stacks[0]'
+	@aws cloudformation describe-stacks --stack-name $(TAG) --region $(REGION) --query 'Stacks[0]'
 
 $(TAG)-workload-bucket: $(BUILD_DIR)/grid_init/init-grid-$(TAG)-workload-bucket
 
 init: $(BUILD_DIR)/grid_init
 
 clean-grid-state:
-	aws --region $(REGION) s3 rm --recursive s3://$(shell aws cloudformation describe-stacks --stack-name $(TAG) --region $(REGION) --query 'Stacks[0].Outputs[2].OutputValue' --output text) || true
+	aws --region $(REGION) s3 rm --recursive s3://$(GRID_TFSTATE_BUCKET) || true
+	@aws s3api delete-objects --bucket $(GRID_TFSTATE_BUCKET) --delete '$(shell aws s3api list-object-versions --bucket $(GRID_TFSTATE_BUCKET) --query='{Objects: Versions[].{Key:Key,VersionId:VersionId}}')' --output text > /dev/null 2>&1 || true
```

Comment:
After the recursive delete and the deletion of the  multiple version, I still see in my buckets still the `deletion` marker.
 * As the bucket is not empty this  triggers an error on  cloudformation